<launch>
    <arg name="message_display" default="screen" doc="output type"/>
    <arg name="whisper_en" default="1" doc="whisper_en"/>
    <arg name="timeout_sec" default="0.3" doc="timeout"/>
    <arg name="main_hz" default="1000" doc="main loop herz"/>
    <arg name="communication" default="rs485" doc="communication"/>
    <arg name="serial_port" default="/dev/recipe.driver.md1k.all" doc="serial_port"/>
    <arg name="baudrate" default="115200" doc="baudrate"/>
    <arg name="inv_motor_in_arr" default="[-1.0, -1.0, -1.0, -1.0]" doc="inv_motor_in_arr"/>
    <arg name="inv_encoder_out_arr" default="[-1.0, -1.0, -1.0, -1.0]" doc="inv_encoder_out_arr"/>
    
    <include file="$(find outtrigger)/launch/base.launch">
        <arg name="WHEEL_NUM" value="4"/>
        <arg name="WHEEL_RADIUS" value="0.1016"/>
        <arg name="WHEEL_LENGTH" value="0.282"/>
        <arg name="WHEEL_WIDTH" value="0.2797"/>
        <arg name="GEAR_RATIO" value="30.0"/>
        <arg name="MOTOR_TICK" value="65535.0"/>
        <arg name="WHEEL_MAX_RPM" value="50.0"/>
        <arg name="MOTOR_CALIBRATION" value="1.0"/>
        <arg name="message_display" value="$(arg message_display)"/>
        <arg name="whisper_en" value="$(arg whisper_en)"/>
        <arg name="timeout_sec" value="$(arg timeout_sec)"/>
        <arg name="main_hz" value="$(arg main_hz)"/>
        <arg name="communication" value="$(arg communication)"/>
        <arg name="serial_port" value="$(arg serial_port)"/>
        <arg name="baudrate" value="$(arg baudrate)"/>
        <arg name="inv_motor_in_arr" value="$(arg inv_motor_in_arr)"/>
        <arg name="inv_encoder_out_arr" value="$(arg inv_encoder_out_arr)"/>
    </include>
</launch>
